ftdetect/gofiletype.vim

```diff
@@ -31,9 +31,23 @@ au BufReadPost *.s call s:gofiletype_post()
 
 au BufRead,BufNewFile *.tmpl set filetype=gohtmltmpl
 
-" make sure we explicitly look for a `go.mod` and the `module` starts from the
-" beginning
-au BufNewFile,BufRead go.mod
-	\ if getline(1) =~ '^module.*' | set filetype=gomod |  endif
+" Set the filetype if the first non-comment and non-blank line starts with
+" 'module <path>'.
+au BufNewFile,BufRead go.mod call s:gomod()
+
+fun! s:gomod()
+  for l:i in range(1, line('$'))
+    let l:l = getline(l:i)
+    if l:l ==# '' || l:l[:1] ==# '//'
+      continue
+    endif
+
+    if l:l =~# '^module .\+'
+      set filetype=gomod
+    endif
+
+    break
```

Comment:
Should this break be moved into the previous `if` block?

Reply:
No, once we see a non-blank line or a line that is not a non-comment we don't need to look any further. `module pkg` must the first line.